src/main/kotlin/org/pkl/lsp/VirtualFileManager.kt

```diff
@@ -17,10 +17,10 @@ package org.pkl.lsp
 
 import java.net.URI
 import java.nio.file.*
-import java.util.*
+import java.util.concurrent.ConcurrentHashMap
 
 class VirtualFileManager(project: Project) : Component(project) {
-  private val files: MutableMap<URI, BaseFile> = Collections.synchronizedMap(WeakHashMap())
+  private val files: MutableMap<URI, BaseFile> = ConcurrentHashMap()
```

Comment:
This was yet another place that was causing us to initialize the stdlib multiple times. The weakmap was kicking out modules (including stdlib modules).

Since modules can point to other modules directly (e.g. via computing a node's type), I'm not really sure if using a weak map is actually the correct thing to do.

This does mean that the server's memory will grow. However, in practice, I'm not sure if this matters all that much. For example, many editor tools will scan and index an entire project and create symbols, which means that they have every project file stored in an internal state like we will here.

Reply:
If this is a concern you could turn this into a `ConcurrentMap<URI, WeakReference<BaseFile>>`, with an adjacent `ConcurrentLinkedQueue<URI>()` to track loading/unloading of file content as needed.